README.md

```diff
@@ -1,20 +1,113 @@
+`npm run lint`: runs `eslint` on the project  
+`npm run lint:fix`: tries to fix all warnings and errors `eslint` returns  
+`npm run dependency-graph`: creates a dependency graph of all files and outputs it into the console. It can be copied to [here](http://www.webgraphviz.com/) to view the graph.
+
+### Linter 
+To keep the code clean, eslint is used as a linter. The [rules](https://eslint.org/docs/latest/rules) for the linter are specified in `.eslintrc`.
+
+### Deployment 
+There is a `dockerfile` to create a docker image out of this application. When starting the container, you have to make sure that you provide the correct [environment variables](#environmental-variables)
+
+## Functionality
+The functionality of the application can be split into the following parts:
+
+1. [Signup](#signup)
+2. [Signin](#signin)
+3. [Establishing a connection to the websocket](#establish-connection-to-websocket)
+
+### Use Case Diagrams
+The following use case diagrams describe how the functionalities depend on each other. They are split into functionalities that can be used through the Rest-API and through the websocket. 
+
+<p align="center">
+  <img src="./documentation/diagramme/usecase_restAPI.drawio.svg" />
+</p>  
+
+
+<p align="center">
+  <img src="./documentation/diagramme/usecase_websocket.drawio.svg" />
+</p>  
+
+### Activitydiagram
+The following Activitydiagrams describe the internal logic of each functionality.
+#### Signup
+```mermaid
+
+
+stateDiagram-v2
+    state if_state <<choice>>
+    state "Client sends username and password to signup endpoint" as send_data
+    state "Server returns authentication token" as send_token
+    state userCheck <<choice>>
+    [*] --> if_state
+    if_state --> send_data
+    userCheck --> if_state : username already exists
+    send_data --> userCheck
+    userCheck --> send_token : username does not exist
+    send_token --> [*]
+```
+
+#### Signin
+```mermaid
+stateDiagram-v2
+    state if_state <<choice>>
+    state "Client sends username and password to signin endpoint" as send_data
+    state "Server returns authentication token" as send_token
+    state credentialsCheck <<choice>>
+    [*] --> if_state
+    if_state --> send_data
+    credentialsCheck --> if_state : Password or username invalid
+    send_data --> credentialsCheck
+    credentialsCheck --> send_token : credentials valid
+    send_token --> [*]
+
+```
+
+#### Establish connection to websocket
+
+```mermaid
+stateDiagram-v2
+    state "Client sends token to api to estabish websocket 
+    connection" as first_request
+    state "API Returns error" as error
+    state "Server establishes websocket connection and saves the user" as successful_signup
+    state token_check <<choice>>
+    [*] --> first_request
+    first_request --> token_check
+    token_check --> error : Invalid token
+    token_check --> successful_signup : Valid token
+    successful_signup --> [*]
+    error --> [*]
+```
+
+## Statediagram of the rules

```

Comment:
Übersetzung ins Deutsche steht weiterhin aus

Reply:
Passend dazu wurde das Issue #15 erstellt, somit in einem weiteren PR erst behandeln